project/Bichen07/README.rst

```diff
@@ -0,0 +1,180 @@
+.. code:: cpp
+
+   struct VoronoiCell {
+       std::vector<Edge> edges;
+       Point centroid;
+   };
+
+   struct VoronoiDiagram {
+       std::vector<VoronoiCell> cells;
+       std::vector<Point> vertices;
+       std::vector<Edge> edges;
+   };
+
+* **Python Integration (pybind11)**: 
+I will provide a Python interface for ease of use and integration with  
+Python-based libraries such as Matplotlib for visualization and NumPy  
+for data manipulation. This interface will allow users to compute  
+Voronoi diagrams from Python without  
+significant performance penalties or memory duplication.  
+.. code:: python
+
+   class VoronoiCell:
+       def __init__(self, edges, centroid):
+           self.edges = edges
+           self.centroid = centroid
+
+
+*****************
+API Description
+*****************
+
+Function in Python
+++++++++++++++++++
+* ``compute()``: Constructs the Voronoi diagram and returns vertices and edges.  
+
+* ``get_cells()``: Returns the Voronoi cells, each corresponding to a seed  
+  point. 
+
+* ``get_edges()``: Returns edges of the diagram, useful for visualization.  
+
+* ``get_vertices()``: Returns the vertices at the intersection of the edges.  
+
+
+*****************
+Engineering Infrastructure
+*****************
+The project will use:
+
+* Unit Testing: C++ unit tests will be created using Google Test for  
+  validating the core computation of the Voronoi diagrams. Python tests will  
+  use pytest to ensure the Python interface behaves correctly and efficiently.  
+
+* Continuous Integration (CI): GitHub Actions will be used to automatically  
+  run tests on each commit, ensuring the correctness of the C++ and  
+  Python integration, as well as performance regressions.  
+
+* Code Documentation: Documentation will be provided in both Doxygen (for C++)  
+  and Sphinx (for Python), ensuring that developers can easily  
+  extend and use the system.
+
+* Unit Testing: C++ unit tests may use Google Test or an alternative framework,  
+  depending on the architecture and complexity of the code.  
+
+* Continuous Integration (CI): 
+  GitHub Actions will be configured to run tests automatically on each commit.  
+  The workflow will include:
+    - Running Google Test (or an alternative) for the C++ core.
+    - Running pytest for the Python integration.
+    - Benchmarking performance to ensure there are no regressions.
+
+*****************
+Schedule
+*****************
+1. Week 1 (10/07): 
+    * Research existing methods (Python libraries like SciPy) and C++  
+      implementations. 
+    * Study Voronoi algorithms (e.g., Fortune's Algorithm) and Python/C++  
+      integration with Pybind11.
+
+2. Week 2 (10/14): 
+    * Design the hybrid architecture, deciding which parts to implement in C++  
+      and Python. 
+    * Set up project structure and basic Python-C++ integration using Pybind11.  
+
+3. Week 3 (10/21): 
+    * Implement core Voronoi diagram algorithm in C++. Test correctness  
```

Comment:
I have not seen your plan for adding automatic testing and CI.  Please make it clear how and when you will do them.